src/sentry/snuba/discover.py

```diff
@@ -1112,10 +1127,60 @@ def find_measurements_min_max(measurements, min_value, max_value, user_query, pa
         min_value = min(min_values) if min_values else None
 
     if max_value is None:
-        max_values = [row[get_function_alias(column)] for column in max_columns]
+        max_values = [
+            row[get_function_alias("max(measurements.{})".format(measurement))]
+            for measurement in measurements
+        ]
         max_values = list(filter(lambda v: v is not None, max_values))
         max_value = max(max_values) if max_values else None
 
+        fences = []
+        for measurement in measurements:
+            if data_filter == "p90":
+                column_name = get_function_alias(
+                    "percentile(measurements.{}, 0.9)".format(measurement)
+                )
+                fences.append(row[column_name])
+
+            if data_filter == "p95":
+                column_name = get_function_alias(
+                    "percentile(measurements.{}, 0.95)".format(measurement)
+                )
+                fences.append(row[column_name])
+
+            if data_filter == "p99":
+                column_name = get_function_alias(
+                    "percentile(measurements.{}, 0.99)".format(measurement)
+                )
+                fences.append(row[column_name])
+
+            if data_filter in ("upper_inner_fence", "upper_outer_fence"):
+                Q1_column_name = get_function_alias(
```

Comment:
Would `p25_column` and `p75_column` work here? Adding another way to refer to those values will be confusing for future us.